tests/pipelines/lock.nf.test

```diff
@@ -0,0 +1,391 @@
+
+            assert iridanext_metadata.containsKey("sample3")
+            assert iridanext_metadata.sample3.metadata_1 == "3.1"
+            assert iridanext_metadata.sample3.metadata_2 == "3.2"
+            assert iridanext_metadata.sample3.metadata_3 == ""
+            assert iridanext_metadata.sample3.metadata_4 == "3.4"
+            assert iridanext_metadata.sample3.metadata_5 == "3.5"
+            assert iridanext_metadata.sample3.metadata_6 == ""
+            assert iridanext_metadata.sample3.metadata_7 == "3.7"
+            assert iridanext_metadata.sample3.metadata_8 == "3.8"
+        }
+    }
+
+    test("Missing column") {
+        tag "pipeline_lock"
+        tag "pipeline_lock_missing_column"
+
+        when {
+            params {
+                input = "$baseDir/tests/data/samplesheets/lock/missing_column.csv"
+                outdir = "results"
+
+                transformation = "lock"
+            }
+        }
+
+        then {
+            assert workflow.success
+            assert path("$launchDir/results").exists()
+
+            // Check Transformation (Machine-Readable)
+            def transformation = path("$launchDir/results/transformation/transformation.csv")
+            assert transformation.exists()
+
+            assert transformation.text.contains("sample,metadata_1,metadata_2,metadata_3,metadata_5,metadata_6,metadata_7,metadata_8")
+            assert transformation.text.contains("sample1,1.1,1.2,1.3,1.5,1.6,1.7,1.8")
+            assert transformation.text.contains("sample2,2.1,2.2,2.3,2.5,2.6,2.7,2.8")
+            assert transformation.text.contains("sample3,3.1,3.2,3.3,3.5,3.6,3.7,3.8")
+
+            // Check Results (Human-Readable)
+            def results = path("$launchDir/results/transformation/results.csv")
+            assert results.exists()
+
+            assert results.text.contains("sample,sample_name,metadata_1,metadata_2,metadata_3,metadata_4,metadata_5,metadata_6,metadata_7,metadata_8")
+            assert results.text.contains("sample1,ABC,1.1,1.2,1.3,,1.5,1.6,1.7,1.8")
+            assert results.text.contains("sample2,DEF,2.1,2.2,2.3,,2.5,2.6,2.7,2.8")
+            assert results.text.contains("sample3,GHI,3.1,3.2,3.3,,3.5,3.6,3.7,3.8")
+
+            // Check IRIDA Next JSON Output
+            def iridanext_json = path("$launchDir/results/iridanext.output.json").json
+            def iridanext_global = iridanext_json.files.global
+            def iridanext_metadata = iridanext_json.metadata.samples
+
+            assert iridanext_global.findAll { it.path == "transformation/results.csv" }.size() == 1
+
+            assert iridanext_metadata.size() == 3
+
+            assert iridanext_metadata.containsKey("sample1")
+            assert iridanext_metadata.sample1.metadata_1 == "1.1"
+            assert iridanext_metadata.sample1.metadata_2 == "1.2"
+            assert iridanext_metadata.sample1.metadata_3 == "1.3"
+            assert iridanext_metadata.sample1.metadata_5 == "1.5"
```

Comment:
Could you also add checks here to make sure `metadata_4` key does not exist?

Reply:
Added in a6d4635